### PR TITLE
Tray: Add left click hide/show feature

### PR DESCRIPTION
**Pull request by 0bCdian**
This pr adds the option to click on the tray to hide or show the window.

I made sure it:

- [x]  adds the corresponding entry to the vesktop settings tab.
- [x]  is optional and disabled by default. 

I'm making this pull request since #262 seems to be stale.

### CODE REVIEW

**Comment by DRAGONTOS** [comment]
+1

File: src/main/mainWindow.ts [hunk 1]
```diff
@@ -73,6 +73,13 @@ const [addSettingsListener, removeSettingsListeners] = makeSettingsListenerHelpe
 const [addVencordSettingsListener, removeVencordSettingsListeners] = makeSettingsListenerHelpers(VencordSettings);
 
 function initTray(win: BrowserWindow) {
+    const trayClickHandler = Settings.store.clickTrayToShowHide
```
Comment (Vendicated) [inline src/main/mainWindow.ts:76]: if you move this ternary inside the callback, it wont require a restart:
```js
const onTrayClick = () => {
	if (Settings.store.clickTrayToShowHide && win.isVisible())
		win.hide();
	else
		win.show();
}
```

Comment (0bCdian) [reply src/main/mainWindow.ts]: Hi! Thanks for reviewing my pr, I have already committed your proposed changes (I don't know how I didn't think of your way of doing it first, it's a much cleaner solution to such a simple feature), sorry for the delay, I don't know how the notification slipped without me noticing.

**Comment by Vendicated** [comment]
thank you!

**Pull request merged**